[Reporting] Reporting info panel touch ups

jloleysens opened the pull request:
## Summary

Closes https://github.com/elastic/kibana/issues/118247

Addresses the points raised the issue and introduces new structure to the flyout. Notable changes:

* Added subsections for the details to live inside of helping the reader scan and find the exact info they are looking for
* Moved warnings and errors to a callout at the top of the flyout. It's important that this information be highlighted here as it is one the main reasons a user might to the flyout.
* Move the report title to the title of the flyout, was "Report info"
* Removed some fields: Export type, Layout, Dimensions (now to separate fields with units denoted), 

## How to test

1. Start Kibana & ES on basic
2. Go to License management and start a trial license
3. Generate a PNG/PDF report in dashboard
4. Go the reporting management UI in Stack Management and view the report flyout by clicking on the name of a report

## Release note

We introduced a new structure to the report details flyout to help readers get to the information they want faster.

## Screenshots

<details>

<summary>Screenshots</summary>

![{D578EF52-04B0-49AE-8610-F30BDDEC6219}](https://user-images.githubusercontent.com/8155004/145866716-2399785d-bde5-4b6d-a88f-1d539b871386.png)

### With warnings

![Screenshot 2021-12-07 at 14 44 17](https://user-images.githubusercontent.com/8155004/145045062-e908ae23-dbf1-40c7-8689-36f5cdad25d4.png)

### With errors

![Screenshot 2021-12-14 at 11 04 29](https://user-images.githubusercontent.com/8155004/145977069-569cf4a6-10bb-4c08-a2ab-cd8fb725e850.png)


### With errors and warnings

![Screenshot 2021-12-07 at 14 39 54](https://user-images.githubusercontent.com/8155004/145045209-30e36cf5-07fd-4170-8e3a-94a8bc0e74e8.png)

</details>


- [x] Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/main/packages/kbn-i18n/README.md)
- [x] [Unit or functional tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html) were updated or added to match the most common scenarios

jloleysens commented:
@elasticmachine merge upstream

elasticmachine commented:
Pinging @elastic/kibana-reporting-services (Team:Reporting Services)

elasticmachine commented:
Pinging @elastic/kibana-app-services (Team:AppServicesUx)

tsullivan commented on x-pack/plugins/reporting/public/management/components/report_info_flyout_content.tsx:
```diff
@@ -20,68 +29,61 @@ const UNKNOWN = i18n.translate('xpack.reporting.listing.infoPanel.unknownLabel',
       }),
       description:
-        info.kibana_name && info.kibana_id ? `${info.kibana_name} (${info.kibana_id})` : NA,
+        info.attempts.toString() +
+        (info.max_attempts
+          ? ' ' +
+            i18n.translate('xpack.reporting.listing.infoPanel.ofMaximumAttempts', {
+              defaultMessage: 'of {maxAttempts}',
+              values: { maxAttempts: info.max_attempts.toString() },
+            })
+          : ''),
```
```suggestion
      description: info.max_attempts
        ? i18n.translate('xpack.reporting.listing.infoPanel.attempts', {
            defaultMessage: '{attempts} of {maxAttempts}',
            values: { attempts: info.attempts, maxAttempts: info.max_attempts },
          })
        : info.attempts,
```

tsullivan commented on x-pack/plugins/reporting/public/management/components/report_info_flyout_content.tsx:
```diff
@@ -95,93 +97,99 @@ export const ReportInfoFlyoutContent: FunctionComponent<Props> = ({ info }) => {
+      title: i18n.translate('xpack.reporting.listing.infoPanel.dimensionsInfoHeight', {
+        defaultMessage: 'Height (in pixels)',
+      }),
+      description: info.layout?.dimensions?.height || UNKNOWN,
```
Let's round the number, since it can be an irrational number
```suggestion
      description:
        info.layout?.dimensions?.height != null
          ? Math.ceil(info.layout.dimensions.height)
          : UNKNOWN,

```

tsullivan commented on x-pack/plugins/reporting/public/management/components/report_info_flyout_content.tsx:
```diff
@@ -95,93 +97,99 @@ export const ReportInfoFlyoutContent: FunctionComponent<Props> = ({ info }) => {
+        defaultMessage: 'Width (in pixels)',
       }),
-      description: info.attempts.toString(),
+      description: info.layout?.dimensions?.width || UNKNOWN,
```
```suggestion
      description:
        info.layout?.dimensions?.width != null ? Math.ceil(info.layout.dimensions.width) : UNKNOWN,

```

tsullivan reviewed (changes requested):
I left a few minor suggestions

jloleysens commented:
@elasticmachine merge upstream

tsullivan reviewed (approved):
LGTM

gchaps commented on x-pack/plugins/reporting/public/management/components/report_info_flyout_content.tsx:
```diff
@@ -95,93 +94,106 @@ export const ReportInfoFlyoutContent: FunctionComponent<Props> = ({ info }) => {
+
+    hasScreenshot && {
+      title: i18n.translate('xpack.reporting.listing.infoPanel.dimensionsInfoHeight', {
+        defaultMessage: 'Height (in pixels)',
```
Are the parens needed?  "Size in bytes" does not contain parens.

gchaps commented on x-pack/plugins/reporting/public/management/components/report_info_flyout_content.tsx:
```diff
@@ -95,93 +94,106 @@ export const ReportInfoFlyoutContent: FunctionComponent<Props> = ({ info }) => {
+    },
+    hasScreenshot && {
+      title: i18n.translate('xpack.reporting.listing.infoPanel.dimensionsInfoWidth', {
+        defaultMessage: 'Width (in pixels)',
```
Same comment about the parens.  Does this work:

> Width in pixels

gchaps commented on x-pack/plugins/reporting/public/management/report_listing.tsx:
```diff
@@ -431,10 +429,7 @@ class ReportListingUi extends Component<Props, State> {
                 defaultMessage: 'Open the Kibana App where this report was generated.',
```
```suggestion
                defaultMessage: 'Open the Kibana app where this report was generated.',
```

gchaps reviewed (commented):
Is it possible to add more spacing between the title "Timestamps" and "Time zone"? It seems a little tight.

jloleysens commented on x-pack/plugins/reporting/public/management/components/report_info_flyout.tsx:
```diff
@@ -56,6 +66,42 @@ export const ReportInfoFlyout: FunctionComponent<Props> = ({ onClose, job }) =>
+      }}
+    >
+      {i18n.translate('xpack.reporting.reportInfoFlyout.openInKibanaAppButtonLabel', {
+        defaultMessage: 'Open in Kibana App',
```
"Open in Kibana" sounds good!

jloleysens commented:
Thanks for the review @gchaps ! I think I've addressed your points.

> Is it possible to add more spacing between the title "Timestamps" and "Time zone"? It seems a little tight.

Great idea, updated, see the screenshot.

gchaps reviewed (approved):
LGTM
